roles/addons/diskless/files/base_module.py

```diff
@@ -0,0 +1,282 @@
+
+        * create_new_image()     -> Define your own image creation process
+        * remove_files()         -> Specify what files to remove when image was properly created
+        * clean()                -> Specify what files to remove when image was not properly created (All possible files)
+
+    You can redefine the following methods when inherit:
+       
+        * generate_files()       -> Contains methods to execute to create image
+        * create_image_folders() -> Create your image folders
+        * generate_file_system() -> Generate your image file system
+
+    Normaly you don't have to redefine other methods in sub classes but also you can for a specific need.
+    Please check all Image class methods that can be redefined before creating a new method in subclass
+    """
+
+    # All type of image has a boot_file_template
+    boot_file_template = ''
+
+    # Each image has it's own base directory in IMAGES_DIRECTORY directory
+    IMAGES_DIRECTORY = '/var/www/html/preboot_execution_environment/diskless/images/'
+
+
+    def __init__(self, name, *args):
+        """Class consructor.
+        The constructor take in argument the name of the image, and the creation arguments (in *args).
+        To get an already existing image you have previously created you must call this constructor whith only image 'name' to load it.
+        To create a new image you must call this constructor with image 'name' and all other class constructor arguments.
+        When you redefine Image class constructor you must define your constructor and create_new_image method as following:
+
+            def __init__(self, name, arg1 = None, arg2 = None, argX = None...): <- You must strictly follow this syntax with your custom args (arg1, arg2, argX ...) 
+                super().__init__(name, arg1, arg2, argX...)                     <-
+                                                                                <-
+            def create_new_image(self, arg1, arg32, argX...):                   <-
+                ...(your code)                                                  <- From here you can custom
+
+        For an exemple look at DemoClass class in demo_module
+
+        :param name: The name of the image
+        :type name: str
+        :param *args: The list of arguments for image creation
+        :type *args: list of arguments
+        """
+
+        # Check name format
+        if len(name.split()) > 1 or not isinstance(name, str):
+            raise ValueError('Invalid name format.')
+
+        # Set up image name
+        self.name = name
+
+        # Set up image directory, it is a mandatory directory for all images
+        self.IMAGE_DIRECTORY = Image.IMAGES_DIRECTORY + self.name
+        
+        # If image already exist, and not all other arguments excepts name are None: 
+        # Bad usage of constructor
+        if os.path.isdir(self.IMAGE_DIRECTORY) and not all(arg is None for arg in args):
```

Comment:
We need to be sure that **not all args are None**,  not if **there is no None in args**.
`None in args` isn't correct because there can be None arguments. But we don't need all arguents as None.

Reply:
With an existing image we must have this type of args:
`args = [None, None, None]`
Not these (Bad constructor usage):
`args = [None, 'something', None]` 
`args = ['something', 'something', 'something']` 


We need to check that all args are `None`, not if there is no `None in args`.

Reply:
I have read it too fast, my proposition is wrong.
sorry, I'm closing this as resolved.